src/HassClient/Client/HassClient.cs

```diff
@@ -292,6 +323,13 @@ public class HassClient : IHassClient
         /// </summary>
         private IClientWebSocket? _ws;
 
+        public IObservable<HassEvent> HassEventsObservable { get; }
+
+        public IObservable<ConnectionStatus> ConnectionStatusObservable { get; }
+
+        private event EventHandler<HassEvent>? HassEvents;
+        private event EventHandler<ConnectionStatus>? ConnectionStautsEvents;
+
         public HassClient(ILoggerFactory? loggerFactory = null) : this(

```

Comment:
This class has a lot of responsibilities now and the API is not very clear. For our purpose in ND it seems OK for now but I think we should split it up some more. Not sure yet exactly how

Reply:
I think the events belong to this class but the connection status and "runtime" should be split. I will think of it. Good point